_posts/pl/2021-09-31-zarzadzanie-stanem-aplikacji-frontendowej.md

```diff
@@ -0,0 +1,225 @@
+---
+layout:     post
+title:      "Zarządzanie stanem aplikacji frontendowej na przykładzie NgRx"
+date:       2021-09-21 6:00:00 +0100
+published:  true
+didyouknow: false
+lang:       pl
+author:     mchrapkowicz
+image:      /assets/img/posts/2021-09-31-zarzadzanie-stanem-aplikacji-frontendowej/ngrx.webp
+description: "W miarę rozwijania złożonych aplikacji webowych ważnym i nieoczywistym zagadnieniem staje się projektowanie przepływu informacji pomiędzy komponentami. W tym artykule poznamy jedną z implementacji architektury Redux wykorzystującej koncepcję globalnego stanu aplikacji oraz pokażemy przykład jej zastosowania w bibliotece NgRx."
+tags:
+- frontend
+- redux
+- state management
+- zarządzanie stanem
+- store
+- reducer
+- action
+- effect
+- angular
+- react
+---
+
+W miarę rozwijania złożonych aplikacji webowych ważnym i nieoczywistym zagadnieniem staje się projektowanie przepływu informacji pomiędzy komponentami. Często mamy do czynienia z wieloma źródłami danych. Mogą to być na przykład najróżniejsze zewnętrzne serwisy czy polecenia wprowadzane przez użytkownika. Podstawowym podejściem jest przekazywanie tych danych przy pomocy rozwiązań dostarczanych przez wykorzystywany framework (na przykład data binding w Angularze), jednak bardzo łatwo można wyobrazić sobie sytuację, w której informacje, z których chcemy skorzystać, znajdują się w zupełnie innej części aplikacji, a przed nami stoi zadanie „przepchania" ich aż do interesującego nas miejsca. Jest to oczywiście rozwiązanie nieskalowalne, trudne w utrzymaniu i niezbyt eleganckie.
+
+Wyobraźmy sobie zatem sytuację, w której wszystkie możliwe dane w sposób uporządkowany trafiają do jednego miejsca i mogą być z niego odczytane w dowolnej chwili. Prawda, że cudna koncepcja? Nazywamy ją "single source of truth", czyli, w wolnym tłumaczeniu, jedno źródło prawdy.  Wykorzystanie takiego pomysłu pozwala pisać bardziej przewidywalny i łatwiej testowalny kod, jak również znacząco zwiększa możliwość skalowania aplikacji. W tym wpisie postaram się nieco przybliżyć jedną z implementacji architektury Redux wykorzystującej koncepcję globalnego stanu aplikacji oraz pokażę przykład jej zastosowania w bibliotece NgRx.
+
```

Comment:
wstęp jest trochę argumentem siły, trochę przez narzucenie "oczywiste jest, że to jest lepsze, estetyczne, skalowalne, itp". czyli trochę "uwierzcie mi, że tak jest".
widzę, że dalej odnosisz się do tego i pogłębiasz argumentację ponad "ja to wiem, słuchajcie mnie". może warto we wstępie poddać w wątpliwość dotychczasowe rozwiązania i napisać "przepychanie parametrów jest niewdzięcznym zadaniem" "debugowanie zmiany stanu z wielu komponentów jest długotrwałe" itp. fakty, spostrzeżenia, nawet pytania "czy to faktycznie najlepszego co możemy zrobić?" albo nawet "nie wydaje się, żeby to był szczyt naszych możliwości". dalej, na przestrzeni artykułu dojdziesz, przykładami i opisem, że redux się lepiej skaluje, lepiej debuguje itd. a takie stwierdzenia "oczywiste jest" "widzimy że" "wiemy że" podasz dopiero w podsumowaniu, gdzie będziesz już miał podparte to wywodem? :)